rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+| FG1 having 3rd file slice in pending compaction and few log files added to latest file slice                                                          | latest file slice should be merged w/ log files from last but one file slice. log files are ordered based on completion times. | latest file slice should be merged w/ log files from last but one file slice. log files are ordered based on log version and write token |                                                                |
+| Time travel Query                                                                                                                                     | works as expected                                                                                                              | Assuming commit time is in active timeline, we are good. Else, LSM timeline read support needs to be ported over.                        |
+| Incremental Query                                                                                                                                     | works as expected                                                                                                              | Assuming commit time is in active timeline, we are good. Else, LSM timeline read support needs to be ported over.                        |                                                                                       |
+
+
+### Table properties
+- payload type inference from payload class need to be ported.
+- hoodie.record.merge.mode needs to be ported over. 
+
+### MDT changes:
+- MDT schema changes need to be ported.
+
+### Log reader/format changes:
+- New log header type need to be ported.
+- For unsupported features, meaningful errors should be thrown. For eg, partial update has been enabled and if those log files are read using 0.16.0 reader, we should fail and throw a meaningful error.
+- For deletion vector, we should add the fallback option to 0.16.0 so that we give up on perf optimization, but still support reads w/o any failures. 
+
+### Read optimized query:
+- No explicit changes required (apart from timeline, and FSV) to support read optimized query.  
+
+### Incremental reads and time travel query: 
+- Incremental reads in 1.x is expected to have some changes and design is not fully out. So, until then we have to wait to design 0.16.x read support for 1.x tables for incremental queries. 
+- Time travel query: I don't see any changes in 1.x wrt time travel query. So, as per master, we are still using commit time(i.e not completed time) to serve time travel query. Until we make any changes to that, we do not need to add any additional support to 0.16.x reader. But if we plan to make changes in 1.x, we might have to revisit this. 
```

Comment:
```suggestion
- Time travel query: There are n't notable changes in 1.x wrt time travel query. We are still using commit time(i.e not completed time) to serve time travel query. Until we make any changes to that, we do not need to add any additional support to 0.16.x reader. But if we plan to make changes in 1.x, we might have to revisit this. 
```

Reply:
we need to do this as well I think. So lets think based on this change happening.